lib/rspec/expectations/fail_with.rb

```diff
@@ -19,18 +19,23 @@ def differ
 
       # Raises an RSpec::Expectations::ExpectationNotMetError with message.
       # @param [String] message
-      # @param [Object] expected
-      # @param [Object] actual
+      # @param [Object] matcher
       #
-      # Adds a diff to the failure message when `expected` and `actual` are
+      # Adds a diff to the failure message when `matcher.expected` and `matcher.actual` are
       # both present.
-      def fail_with(message, expected=nil, actual=nil)
+      def fail_with(message, matcher=nil)
         unless message
           raise ArgumentError, "Failure message is nil. Does your matcher define the " \
                                "appropriate failure_message[_when_negated] method to return a string?"
         end
 
-        message = ::RSpec::Matchers::ExpectedsForMultipleDiffs.from(expected).message_with_diff(message, differ, actual)
+        message = if !matcher.nil?
```

Comment:
As you can see above we already inject the differ, it just can't come from a param here, or at least, it'd have to be an extra param.